test/e2e/suite/certificates/othernamesan.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2020 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certificates
+
+import (
+	"context"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"encoding/pem"
+	"time"
+
+	"github.com/cert-manager/cert-manager/e2e-tests/framework"
+	e2eutil "github.com/cert-manager/cert-manager/e2e-tests/util"
+	cmapi "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	cmmeta "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
+	"github.com/cert-manager/cert-manager/test/unit/gen"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = framework.CertManagerDescribe("othername san processing", func() {
+
+	const (
+		testName      = "test-othername-san-processing"
+		issuerName    = "certificate-othername-san-processing"
+		secretName    = testName
+		nameTypeEmail = 1
+	)
+
+	var (
+		oidExtensionSubjectAltName = asn1.ObjectIdentifier{2, 5, 29, 17}
+		emailAddresses             = []string{"email@domain.com"}
+	)
+	// StringValueLikeType type for asn1 encoding. This will hold
+	// our utf-8 encoded string.
+	type StringValueLikeType struct {
+		A string `asn1:"utf8"`
+	}
+
+	type OtherName struct {
+		OID   asn1.ObjectIdentifier
+		Value StringValueLikeType `asn1:"tag:0"`
+	}
+
+	f := framework.NewDefaultFramework("certificate-othername-san-processing")
+	createCertificate := func(f *framework.Framework, OtherNameSANs []cmapi.OtherNameSAN) (*cmapi.Certificate, error) {
+		crt := &cmapi.Certificate{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: testName + "-",
+				Namespace:    f.Namespace.Name,
+			},
+			Spec: cmapi.CertificateSpec{
+				SecretName: secretName,
+				PrivateKey: &cmapi.CertificatePrivateKey{RotationPolicy: cmapi.RotationPolicyAlways},
+				IssuerRef: cmmeta.ObjectReference{
+					Name: issuerName, Kind: "Issuer", Group: "cert-manager.io",
+				},
+				OtherNameSANs:  OtherNameSANs,
+				EmailAddresses: emailAddresses,
```

Comment:
nitpick: I like that we test email address here, but could we test all other fields of the SAN too? Would be a shame to miss if we introduced a bug with, say, DNS names or URIs but not for emails.